File: algorithm/shortest-palindrome.go
func shortestPalindrome(s string) string {
	if len(s) <= 1 {
		return s
	}
	for i := len(s) - 1; i >= 0; i-- {
		if isPalindrome(s[:i+1]) {
			return s[i+1:] + s
		}
	}
	return ""
}

func isPalindrome(s string) bool {
	for i := 0; i < len(s)/2; i++ {
		if s[i] != s[len(s)-1-i] {
			return false
		}
	}
	return true
}

File: algorithm/shortest-palindrome.ts
function shortestPalindrome(s: string): string {
    let i = 0;
    let j = s.length - 1;
    while (i < j) {
        if (s[i] === s[j]) {
            i++;
            j--;
        } else {
            break;
        }
    }
    if (i >= j) {
        return s + s.slice(0, i).split("").reverse().join("");
    }
    let p = s.slice(0, i);
    let q = s.slice(i, j + 1);
    return shortestPalindrome(q) + q + p;
};